pooch/downloaders.py

```diff
@@ -308,3 +316,113 @@ def callback(data):
             ftp.quit()
             if ispath:
                 output_file.close()
+
+
+if paramiko is not None:
```

Comment:
This check would be better in `SFTPDownloader.__ini__` so that the class always exists (which simplifies a lot of the other code in this PR).

Reply:
I inserted the following under SFTPDownloader.__init__
```suggestion
        if paramiko is None:
            raise ImportError(
                'SFTP is only supported if paramiko is installed'
            )
```